dio/test/encoding_test.dart

```diff
@@ -1,90 +1,129 @@
+      final result =
+          'a=%E4%BD%A0%E5%A5%BD&b=5%7C6&c%5Bd%5D=8&c%5Be%5D%5Ba%5D=5&c%5Be%5D%5Bb%5D=foo%2Cbar&c%5Be%5D%5Bc%5D=foo+bar&c%5Be%5D%5Bd%5D%5B%5D=foo&c%5Be%5D%5Bd%5D%5B%5D=bar&c%5Be%5D%5Be%5D=foo%5Ctbar';
+      expect(
+        Transformer.urlEncodeMap(
+          {
+            'a': '你好',
+            'b': ListParam<int>([5, 6], ListFormat.pipes),
+            'c': {
+              'd': 8,
+              'e': {
+                'a': 5,
+                'b': ListParam<String>(['foo', 'bar'], ListFormat.csv),
+                'c': ListParam<String>(['foo', 'bar'], ListFormat.ssv),
+                'd': ListParam<String>(['foo', 'bar'], ListFormat.multi),
+                'e': ListParam<String>(['foo', 'bar'], ListFormat.tsv),
+              },
+            },
+          },
+          ListFormat.multiCompatible,
+        ),
+        result,
+      );
+    });
   });
 
-  test('custom', () {
-    //a=你好&b=5|6&c[d]=8&c[e][a]=5&c[e][b]=foo,bar&c[e][c]=foo+bar&c[e][d][]=foo&c[e][d][]=bar&c[e][e]=foo\tbar
-    final result =
-        'a=%E4%BD%A0%E5%A5%BD&b=5%7C6&c%5Bd%5D=8&c%5Be%5D%5Ba%5D=5&c%5Be%5D%5Bb%5D=foo%2Cbar&c%5Be%5D%5Bc%5D=foo+bar&c%5Be%5D%5Bd%5D%5B%5D=foo&c%5Be%5D%5Bd%5D%5B%5D=bar&c%5Be%5D%5Be%5D=foo%5Ctbar';
-    expect(
-      Transformer.urlEncodeMap(
-        {
-          'a': '你好',
-          'b': ListParam<int>([5, 6], ListFormat.pipes),
-          'c': {
-            'd': 8,
-            'e': {
-              'a': 5,
-              'b': ListParam<String>(['foo', 'bar'], ListFormat.csv),
-              'c': ListParam<String>(['foo', 'bar'], ListFormat.ssv),
-              'd': ListParam<String>(['foo', 'bar'], ListFormat.multi),
-              'e': ListParam<String>(['foo', 'bar'], ListFormat.tsv),
+  group(Transformer.urlEncodeQueryMap, () {
+    test(ListFormat.csv, () {
+      expect(
+        Transformer.urlEncodeQueryMap({
+          'foo': ListParam(['1', '%', '\$'], ListFormat.csv)
+        }),
+        'foo=1,%25,%24',
+      );
+    });
+
+    test('custom', () {
+      expect(
+        Transformer.urlEncodeQueryMap(
+          {
+            'a': '你好',
+            'b': ListParam<int>([5, 6], ListFormat.pipes),
+            'c': {
+              'd': 8,
+              'e': {
+                'a': 5,
+                'b': ListParam<Object>(['foo', 'bar', 1, 2.2], ListFormat.csv),
+                'c': ListParam<Object>(['foo', 'bar', 1, 2.2], ListFormat.ssv),
+                'd':
+                    ListParam<Object>(['foo', 'bar', 1, 2.2], ListFormat.multi),
+                'e': ListParam<Object>(['foo', 'bar', 1, 2.2], ListFormat.tsv),
+              },
             },
           },
-        },
-        ListFormat.multiCompatible,
-      ),
-      result,
-    );
+          ListFormat.multiCompatible,
+        ),
+        'a=%E4%BD%A0%E5%A5%BD&b=5|6&c[d]=8&c[e][a]=5&c[e][b]=foo,bar,1,2.2&c[e][c]=foo%20bar%201%202.2&c[e][d][]=foo&c[e][d][]=bar&c[e][d][]=1&c[e][d][]=2.2&c[e][e]=foo\\tbar\\t1\\t2.2',
```

Comment:
I have no idea what the correct encoding for TSV values looks like, this is probably wrong.
Additionally there are 2 ways to encode SSV (`%20`, `+`) which we have no way to separate atm.
Everything else seems fine.